web/projects/ui/src/app/services/config.service.ts

```diff
@@ -57,24 +56,60 @@ export class ConfigService {
     return window.isSecureContext || this.isTor()
   }
 
-  isLaunchable(
-    state: PackageState,
-    status: PackageMainStatus,
-    interfaces: Record<string, InterfaceDef>,
-  ): boolean {
+  isLaunchable(state: PackageState, status: PackageMainStatus): boolean {
     return (
-      state === PackageState.Installed &&
-      status === PackageMainStatus.Running &&
-      hasUi(interfaces)
+      state === PackageState.Installed && status === PackageMainStatus.Running
     )
   }
 
-  launchableURL(pkg: PackageDataEntry): string {
-    if (!this.isTor() && hasLocalUi(pkg.manifest.interfaces)) {
-      return `https://${lanUiAddress(pkg)}`
+  /** ${scheme}://${username}@${host}:${externalPort}${suffix} */
+  launchableAddress(
+    interfaces: InstalledPackageDataEntry['service-interfaces'],
+  ): string {
+    const ui = Object.values(interfaces).find(i => i.type === 'ui')
+
+    if (!ui) return ''
+
+    const host = ui.hostInfo
+    const addressInfo = ui.addressInfo
+    const scheme = this.isHttps() ? 'https' : 'http'
+    const username = addressInfo.username ? addressInfo.username + '@' : ''
+    const suffix = addressInfo.suffix ? '/' + addressInfo.suffix : ''
+    const url = new URL(`${scheme}://${username}placeholder${suffix}`)
+
+    if (host.kind === 'multi') {
+      const onionHostname = host.hostnames.find(
+        h => h.kind === 'onion',
+      ) as HostnameInfoOnion
+
+      if (this.isTor() && onionHostname) {
+        url.hostname = onionHostname.hostname.value
+      } else {
+        const ipHostname = host.hostnames.find(
```

Comment:
Just taking the first ip hostname you find? Not trying to match it to the address they're currently on?

Reply:
Look at line 94. Just trying to figure out if they have an IP hostname. This won't work once we have domains.

Reply:
Hmm, ok, I guess that works